### DISCUSSION TITLE
Pyright's language server doesn't accept `textDocument/didOpen`?

### DISCUSSION THREAD

**Discussion by decorator-factory** [microsoft/pyright#2632] [General]
I'm trying to work with the Pyright language server. I'm sending this message after the `initialize` message got a response:
```js
        {
            "jsonrpc": "2.0",
            "id": 6,
            "method": "textDocument/didOpen",
            "params": {
                "textDocument": {
                    "uri": "<uri of a file>",
                },
            },
        }
```
But I get this in response:
```js
{'jsonrpc': '2.0', 'id': 6, 'error': {'code': -32601, 'message': 'Unhandled method textDocument/didOpen'}}
```
What am I doing wrong? The language server included `'textDocumentSync': 2` in its capabilities, so shouldn't `textDocument/didOpen` work?

I got quite overwhelmed by the LSP specification... So I'm not sure how to even debug this.

**Comment by erictraut** [comment]
I would expect this to work. Here's where pyright's language server handles this message: https://github.com/microsoft/pyright/blob/cae15781a3049d021133e7ca3ca2b85527faab71/packages/pyright-internal/src/languageServerBase.ts#L813.

**Reply by DanielCodesphere** [reply]
The issue seems to be that `textDocument/didOpen` is a notification and not a request, so it doesn't have an id.
Without the id it should work.
